### PR TITLE
Check $FLOW_PATH in environment variables and check `which flow`

### PR DESCRIPTION
**Pull request by tayloris**
…path

*Insert a description of your pull request (PR) here, and check off the boxes below when they are done.*

---

### Contributor checklist

- [x] :tada: This PR closes ##254
- [x] :scroll: I have broken down my PR into the following tasks:
   - [x] Check if $FLOW_PATH is set, if so, use that. 
   - [x] If $FLOW_PATH points to something that doesn't exist, fail hard.
   - [x] Check $PATH for flow, 
       - [x] if $PATH for flow not found, fail hard with a message that flow was not found
       - [x] If FAIL include a message that either flow needs to be installed (link?) 
       - [x] or the user needs to set $FLOW_PATH manually

### CODE REVIEW

File: src/flownet/ert/_flow_job.py [hunk 1]
```diff
@@ -26,8 +27,27 @@ def run_flow():
...
-    subprocess.run(
-        [os.environ.get("FLOW_PATH", "/usr/bin/flow"), args.data_file], check=True
-    )
+    if "FLOW_PATH" in os.environ:
+        flow_path = os.environ.get("FLOW_PATH")
+        if os.path.isfile(flow_path):
+            # Runing flow from ENV variable
+            subprocess.run([flow_path, args.data_file], check=True)
+        else:
+            raise AssertionError("FLOW_PATH points to a path that doesn't exist")
```
Comment (anders-kiaer) [inline src/flownet/ert/_flow_job.py:36]:
```suggestion
            raise AssertionError("FLOW_PATH does not point at a file that exists")
```

File: src/flownet/ert/_flow_job.py [hunk 2]
```diff
@@ -26,8 +27,27 @@ def run_flow():
...
+    elif shutil.which("flow") is None:
+        raise AssertionError(
+            "OPM/flow is not installed.\
+                             Follow instructions in https://opm-project.org/ to install flow."
+        )
```
Comment (anders-kiaer) [inline src/flownet/ert/_flow_job.py:41]: Mention here that _if_ it is already installed, the user should make sure it is available in `$PATH` (or alternatively use the environment variable `$FLOW_PATH`).

File: src/flownet/ert/_flow_job.py [hunk 3]
```diff
@@ -26,8 +27,27 @@ def run_flow():
...
+    else:
+        flow_path = shutil.which("flow")
+        if os.path.isfile(flow_path):
```
Comment (anders-kiaer) [inline src/flownet/ert/_flow_job.py:44]: I think we don't need this check. If `shutil.which("flow") is not None` (which is the case here) it should already point to a file?

**Review by anders-kiaer — CHANGES_REQUESTED** [review]
LGTM.

Maybe restructure the code slightly? Currently `subprocess.run(...)` is one multiple lines.

Could instead 
1) First find a valid path to `flow` executable (and store it in a variable), and do any error raising.
2) Run subprocess with the found `flow` executable.

This will also move some code out of the `if/else` blocks.

File: src/flownet/ert/_flow_job.py (hunk 3 above)
Comment (tayloris) [reply src/flownet/ert/_flow_job.py:44]: Yes, thats true

**Comment by tayloris** [comment]
I included comments from  @anders-kiaer . The entry point to ERT is defined in `setup.py` so I think the best place to inform the user about  $FLOW_PATH is in the `README` for flownet installation. The implementation will raise an error informing about $FLOW_PATH if there is any problem when we want to run flow. 

If we want to raise a warning about $FLOW_PATH  we could do it in main() but I'm not sure about it. Let me know what you think.

File: src/flownet/ert/_flow_job.py [hunk 4]
```diff
@@ -26,8 +27,22 @@ def run_flow():
...
+            )
+    elif flow_path is None:
+        raise AssertionError(
+            r"OPM/flow is not installed.\n \
+            Follow instructions in https://opm-project.org/ to install flow.\n \
+            If OPM/flow it is already installed, make sure it is available in $PATH,\n \
+            or alternatively use the environment variable $FLOW_PATH"
+        )
+
+    subprocess.run([flow_path, args.data_file], check=True)
```
Comment (wouterjdb) [inline src/flownet/ert/_flow_job.py:46]:
```suggestion
    flow_path = shutil.which("flow")
    if "FLOW_PATH" in os.environ:
        flow_path = os.environ.get("FLOW_PATH")
        if not os.path.isfile(flow_path):
            raise FileNotFoundError(
                r"$FLOW_PATH does not point at a file that exists.\n \
                Please, use the environment variable $FLOW_PATH to indicate a path for OPM Flow"
            )
    elif flow_path is None:
        raise RuntimeError(
            r"OPM Flow could not be found.\n \
            Follow instructions on https://opm-project.org/ to install OPM Flow.\n \
            If OPM Flow it is already installed, make sure it is available in $PATH,\n \
            or alternatively use the environment variable $FLOW_PATH."
        )

    subprocess.run([flow_path, args.data_file], check=True)
```

**Review by wouterjdb — CHANGES_REQUESTED** [review]
LGTM, minor suggestions.

**Review by wouterjdb — APPROVED** [review]
LGTM 👍

**Review by anders-kiaer — APPROVED** [review]

**Pull request merged**